azerothcore/azerothcore-wotlk
# fix (core/logging/conf): Fix Logging for Modules

## Changes Proposed:
-  Fix custom module logging and appender. Also included Logging Async Conf.
![image](https://user-images.githubusercontent.com/16887899/166328068-a6cf8968-25f9-4079-bc4c-96d06f10f4cf.png)
![image](https://user-images.githubusercontent.com/16887899/166328106-d68630b8-6b3c-4185-a846-a2f9461189b1.png)

Partial TC Cherrypick: https://github.com/TrinityCore/TrinityCore/commit/d1594998f80762fa58f64cf123f9bf9cb77036e4
Co-Authored-By: leak [leak@users.noreply.github.com](mailto:leak@users.noreply.github.com)

TC Cherrypick: https: //github.com/TrinityCore/TrinityCore/commit/c71987b1a1403fa20654cc24b37448ca807ff363
Co-Authored-By: Shauren <shauren.trinity@gmail.com>

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11297

## SOURCE:


## Tests Performed:

- Builds
- Runs
- Custom Module Logging now works


## How to Test the Changes:


1. compile with anticheat module.

2. Set in the world.conf logger.module to 

``` Logger.module=4,Console Server Anticheat```
Then add
```Appender.Anticheat=2,4,15,anticheat.log,a```

3. Anticheat log file is generate but this times now fills in.

4. adding conf and setting to true 
```Log.Async.Enable = 1```
fill allow all logging to be filled async instead of immediately.

5. Also observe other .log files are still logging as intended and that there is no regression.

6. Doing a .reload conf also activately reloads the conf for the modules as well. You will see a confirmation message on the world console as well from the anticheat module.

7. Shut down world. Notice world shutdowns alot faster now as well.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
